### DISCUSSION TITLE
Add header to request

### DISCUSSION THREAD

**Discussion by franklinvillalobos** [adonisjs/core#4287] [Help] [answered]
Hello, I want to know if it is possible to add a header to all request received through midleware.

That is, I want to make a global middleware that adds an additional header before going to route.

I hope you understand my problem.

**Comment by RoyalVICTOR-pro** [comment]
Hi !
Why do you want to do that ? 
If you want to add a header to the response, it's possible but It is not recommended to modify the request header; the client should be the one sending the request.

**Reply by franklinvillalobos** [reply]
I want to use auth with token, but I will receive the token from my frontend as a cookie.
I want to get that cookie and send it to the auth middelware as an Authorization Bearer header as I have already done in Laravel.

**Comment by tomgobich** [comment, marked as answer]
Yep, you can reach through to the underlying request object and set it directly. Here's an example piggybacking off the `SilentAuth` middleware.
```ts
export default class SilentAuthMiddleware {
  /**
   * Handle request
   */
  public async handle({ auth, request }: HttpContextContract, next: () => Promise<void>) {
    /**
     * Check if user is logged-in or not. If yes, then `ctx.auth.user` will be
     * set to the instance of the currently logged in user.
     */

    // try to get token from cookies, if found, set authorization
    const token = request.cookie('auth-session')
    if (token) {
      request.request.headers['authorization'] = `Bearer ${token}`
    }

    await auth.check()
    await next()
  }
}
```

Also note that if you're able to use cookies, then just using session auth instead of token-based auth would be an easier solution.

**Reply by franklinvillalobos** [reply]
Thank you so much! It has worked correctly for me.
